main/src/addins/MonoDevelop.DesignerSupport/MonoDevelop.DesignerSupport/NativePropertyEditor/ComponentModelEditorProvider.cs

```diff
@@ -0,0 +1,136 @@
+//
+// PropertyPadEditorProvider.cs
+//
+// Author:
+//       jmedrano <josmed@microsoft.com>
+//
+// Copyright (c) 2018 
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#if MAC
+
+using System.Linq;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Xamarin.PropertyEditing;
+using MonoDevelop.Core;
+
+namespace MonoDevelop.DesignerSupport
+{
+	class ComponentModelEditorProvider
+	: IEditorProvider
+	{
+		public IReadOnlyDictionary<Type, ITypeInfo> KnownTypes {
+			get;
+		} = new Dictionary<Type, ITypeInfo> ();
+
+		object [] providers;
+
+		public Task<IObjectEditor> GetObjectEditorAsync (object item)
+		{
+			if (item is ComponentModelTarget propertyPadItem) {
+				//on each editor we store current used providers
+				providers = propertyPadItem.Providers ?? Array.Empty<object> ();
+				return Task.FromResult<IObjectEditor> (new ComponentModelObjectEditor (propertyPadItem));
+			}
+			throw new ArgumentException (string.Format ("Cannot get editor for type {0} only ComponentModelObjectEditor types are allowed", item.GetType ().Name));
```

Comment:
This seems wrong. It means that instead of a faulted task, we're going to just synchronously throw.

Most likely this should be:
```suggestion
			return Task.FromException (new ArgumentException (string.Format ("Cannot get editor for type {0} only ComponentModelObjectEditor types are allowed", item.GetType ().Name));
```

Reply:
Fixed